app/src/main/java/com/github/onedirection/Event.java

```diff
@@ -0,0 +1,103 @@
+package com.github.onedirection;
+
+class Event {
+
+    final private int id;
+    final private String name;
+    final private String location;
+    final private String date;
+    final private String start_time;
+    final private String end_time;
+
+
+    public Event(int id,String name, String location, String date, String start_time, String end_time){
+        if(name == null || location == null || date == null || start_time == null || end_time == null){
+            throw new IllegalArgumentException();

```

Comment:
It would be (imo) cleaner to do
```suggestion
        Objects.requireNonNull(name);
        Objects.requireNonNull(location);
        ...
        Objects.requireNonNull(end_time);

```
(Note: we could even write an utility function to write)
```suggestion
        requireNoneNull(name , location, date, start_time, end_time);
```